host/tests/unit/search-index-test.ts

```diff
@@ -532,4 +532,105 @@ module('Unit | search-index', function () {
     });
     assert.ok(definition, 'got definition');
   });
+
+  test('directories do not list entries that match patterns in ignore files', async function (assert) {
+    let realm = TestRealm.create({
+      'sample-post.json': {},
+      'posts/1.json': {},
+      'posts/nested.gts': `
+        import { Card } from 'https://cardstack.com/base/card-api';
+        export class Nested extends Card {}
+      `,
+      'posts/2.json': {},
+      'post.gts': `
+        import { Card } from 'https://cardstack.com/base/card-api';
+        export class Post extends Card {}
+      `,
+      'dir/post-1.json': {},
+    });
+
+    const cardRef = {
+      type: 'exportedCard',
+      module: 'posts/nested.gts',
+      name: 'Nested',
+    };
+
+    const listings = [
+      {
+        kind: 'file',
+        name: 'sample-post.json',
+      },
+      {
+        kind: 'directory',
+        name: 'posts',
+      },
+      {
+        kind: 'file',
+        name: 'post.gts',
+      },
+      {
+        kind: 'directory',
+        name: 'dir',
+      },
+    ] as any;
+
+    const ignoreFile = {
+      name: '.gitignore',
+      kind: 'file',
+    };
+
+    const nestedListings = [
+      {
+        name: '1.json',
+        kind: 'file',
+      },
+      {
+        name: 'nested.gts',
+        kind: 'file',
+      },
+      {
+        name: '2.json',
+        kind: 'file',
+      },
+    ] as any;
+
+    let indexer = realm.searchIndex;
+    await indexer.run();
+    let definition = await indexer.typeOf(cardRef as any);
+    assert.ok(definition, 'got definition');
+
+    let entries = await indexer.directory(new URL(realm.url));
+    assert.deepEqual(entries, listings, 'top level entries are correct');
+    let nestedEntries = await indexer.directory(new URL('posts/', realm.url));
+    assert.deepEqual(
+      nestedEntries,
+      nestedListings,
+      'nested entries are correct'
+    );
+
+    await realm.write('.gitignore', '*.json\n/dir');
+    await indexer.run();
```

Comment:
I had to run the indexer again after doing the update, since `isIgnored` is only checked during `visitDirectory`. should there be changes to the indexer's `update` function instead?

Reply:
probably we need this in both places